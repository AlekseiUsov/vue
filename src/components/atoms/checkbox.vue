<template>
    <label :class="$style.label">
        <input type="checkbox" :class="$style.inputCheckbox" />
        <span :class="$style.checkbox"></span>
        <span :class="$style.checkboxText"></span>
    </label>
</template>

<script>
export default {
    name: "checkbox"
};
</script>

<style lang="scss" module>
@import "@/assets/main.scss";

.label {
    display: flex;
    align-items: center;
}

.inputCheckbox {
    opacity: 0;
}

.checkboxText {
    color: $brown;
}

.checkbox {
    display: block;
    width: 1rem;
    height: 1rem;
    border: 1px solid $yellow;
    border-radius: 0.2em;
    transition: 0.2s;
    margin-right: 0.5rem;
}

.inputCheckbox:checked+.checkbox {
    background-image: url('../../assets/checked.svg');
    background-size: contain;
    background-color: $yellow;
}
</style>