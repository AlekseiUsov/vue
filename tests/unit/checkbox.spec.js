import { mount } from "@vue/test-utils";
import checkbox from "../../src/components/atoms/ToDoCheckbox.vue";

describe("ToDoCheckbox.vue", () => {
  it("тестируем чекбокс", async () => {
    const wrapper = mount(checkbox);
    expect(wrapper).toMatchSnapshot();
  });

  it("тестируем пропсы чекбокс", async () => {
    const wrapper = mount(checkbox, {
      propsData: {
        isChecked: true,
      },
    });
    expect(wrapper.find("input").element.checked).toBe(true);
  });

  it("тестируем клик по чекбоксу", async () => {
    const wrapper = mount(checkbox);

    await wrapper.find("input").trigger("click");
    expect(wrapper.emitted("checkCheckbox")).toBeTruthy();
  });
});
