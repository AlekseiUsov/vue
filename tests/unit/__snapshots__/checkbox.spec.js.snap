// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoCheckbox.vue тестируем клик по чекбоксу 1`] = `<label><input type="checkbox"> <span></span></label>`;

exports[`ToDoCheckbox.vue тестируем чекбокс 1`] = `<label><input type="checkbox"> <span></span></label>`;
